Userland/Libraries/LibCore/IODevice.cpp

```diff
@@ -61,6 +61,32 @@ int IODevice::read(u8* buffer, int length)
     return read_buffer.size();
 }
 
+size_t IODevice::copy_from(const IODevice& source, size_t buffer_size)
+{
+#ifndef __serenity__
+    (void)source;
+    (void)buffer_size;
+    return 0;
+#else
+    if (!source.is_readable())
```

Comment:
Is there a reason for doing the validation here instead of letting the kernel reject it? Is there some common error path that you are trying to optimize for?